File: plugins/afore/src/main/java/eu/geekhome/aforeplugin/AforePlugin.java
package eu.geekhome.aforeplugin;

import com.geekhome.common.extensibility.PluginMetadata;
import com.geekhome.common.localization.Resource;
import eu.geekhome.services.hardware.HardwareAdapterFactory;
import eu.geekhome.services.hardware.HardwarePlugin;
import org.pf4j.*;

public class AforePlugin extends HardwarePlugin implements PluginMetadata {

    private final AforeAdapterFactory _factory;

    public AforePlugin(PluginWrapper wrapper) {
        super(wrapper);
        _factory = new AforeAdapterFactory();
    }

    @Override
    public HardwareAdapterFactory getFactory() {
        return _factory;
    }

    @Override
    public void start() {
        System.out.println("Starting AFORE plugin");
    }

    @Override
    public void stop() {
        System.out.println("Stopping AFORE plugin");
    }

    @Override
    public Resource getName() {
        return R.plugin_name;
    }

    @Override
    public Resource getDescription() {
        return R.plugin_description;
    }
}

File: plugins/afore/src/main/java/eu/geekhome/aforeplugin/AforePlugin.kt
package eu.geekhome.aforeplugin

import eu.geekhome.domain.extensibility.PluginMetadata
import org.pf4j.PluginWrapper
import eu.geekhome.domain.hardware.HardwarePlugin
import eu.geekhome.domain.langateway.LanGatewayResolver
import eu.geekhome.data.localization.Resource
import eu.geekhome.domain.hardware.HardwareAdapter

class AforePlugin(
    wrapper: PluginWrapper,
    private val lanGatewayResolver: LanGatewayResolver) : HardwarePlugin(wrapper), PluginMetadata {

    override fun createAdapters(): List<HardwareAdapter<*>> {
        val result = ArrayList<HardwareAdapter<*>>()
        val adapter = AforeAdapter(pluginId, lanGatewayResolver)
        result.add(adapter)
        return result
    }

    override fun start() {
        println("Starting AFORE plugin")
    }

    override fun stop() {
        println("Stopping AFORE plugin")
    }

    override val name: Resource = R.plugin_name
    override val description: Resource = R.plugin_description
}